astroid/interpreter/_import/spec.py

```diff
@@ -291,9 +291,10 @@ def _precache_zipimporters(path=None):
 def _search_zip(modpath, pic):
     for filepath, importer in list(pic.items()):
         if importer is not None:
-            found = importer.find_module(modpath[0])
+            found = getattr(importer, "find_spec", importer.find_module)(modpath[0])
             if found:
-                if not importer.find_module(os.path.sep.join(modpath)):
+                name = os.path.sep.join(modpath)
```

Comment:
I found that the name `name` is not well suited here because the variable is definitely more a path than a name. Maybe something along `module_path`?

Reply:
That's a good point, I've updated the variable name.